sherpa/include/sherpa/astro/xspec_extension.hh

```diff
@@ -61,6 +63,10 @@ namespace sherpa { namespace astro { namespace xspec {
 
 typedef sherpa::Array< float, NPY_FLOAT > FloatArray;
 typedef float FloatArrayType;
+typedef void (*XSpecFuncRef)( float* ear, int* ne, float* param, int* ifl, float* photar, float* photer );
+typedef void (*XSpecFuncVal)( float* ear, int ne, float* param, const char* filenm, int ifl, float* photar, float* photer );
+typedef void (*XSpecFuncDouble)( const double* energy, int nFlux, const double* params, int spectrumNumber, double* flux, double* fluxError, const char* initStr );
+
```

Comment:
So, XSPEC does define these, but you'd need something like the following untested code

```
#ifdef XSPEC_12_12_0
#include "XSFunctions/Utilities/funcType.h"
#else
#include "funcType.h"
#endif
```

which defines (not all of which we care about):

```
// standard XSPEC11 fortran 77 calls
        typedef void (xsf77Call) (const float* energyArray,
                                  const int& Nenergy,
                                  const float* parameterValues,
                                  const int& spectrumNumber,
                                  float* flux,
                                  float* fluxError);
// double precision fortran 77 calls
        typedef void (xsF77Call) (const double* energyArray,
                                  const int& Nenergy,
                                  const double* parameterValues,
                                  const int& spectrumNumber,
                                  double* flux,
                                  double* fluxError);
// C language calls using C arrays
        typedef void (xsccCall)   (const Real* energyArray,
                                   int Nenergy,
                                   const Real* parameterValues,
                                   int spectrumNumber,
                                   Real* flux,
                                   Real* fluxError,
                                   const char* initString);
// C++ call with initialization string passed.
        typedef void (XSCCall) (const RealArray& energyArray,
                                const RealArray& parameterValues,
                                int spectrumNumber,
                                RealArray& flux,
                                RealArray& fluxError,
                                const std::string& initString);
```

The main issue is the difference on whether arguments are labelled `const` or not.